src/planwise/client/scenarios/handlers.cljs

```diff
@@ -54,12 +54,11 @@
  (fn [{:keys [db]} [_ scenario]]
    (let [current-scenario (get-in db [:scenarios :current-scenario])
          should-update    (= (:id current-scenario) (:id scenario))]
-     (merge (cond
-              should-update {:db (-> db
-                                     (assoc-in [:scenarios :current-scenario]
-                                               (merge current-scenario (select-keys scenario demand-fields))))}
-              :else {})
-            (dispatch-track-demand-information-if-needed scenario)))))
+     (if should-update
+       (merge {:db (-> db (assoc-in [:scenarios :current-scenario]
+                                    (merge current-scenario (select-keys scenario demand-fields))))}
+              (dispatch-track-demand-information-if-needed scenario))
+       {}))))
```

Comment:
No need to return an empty map if the handler shouldn't produce an effect. You can use `when` instead of `if`.